commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/StrippedConcurrentLinkedDeque.java

```diff
@@ -0,0 +1,725 @@
+package org.infinispan.commons.util.concurrent.jdk8backported;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/*
+ * Written by Doug Lea and Martin Buchholz with assistance from members of
+ * JCP JSR-166 Expert Group and released to the public domain, as explained
+ * at http://creativecommons.org/publicdomain/zero/1.0/
+ */
+/**
+ * This is a subset of the java.util.concurrent.ConcurrentLinkedDeque to only have
+ * methods for maintaining a doubly linked list but without maintaining the 
+ * java.util.Collection or java.util.Deque contracts.  This is useful to keep a linked
+ * list that will never be iterated upon but instead will have references to elements
+ * directly in the list that will be removed or moved as needed.
+ * @author wburns
+ * @since 7.1
+ */
+class StrippedConcurrentLinkedDeque<E> {
+   /**
+    * A node from which the first node on list (that is, the unique node p
+    * with p.prev == null && p.next != p) can be reached in O(1) time.
```

Comment:
This was how the ConcurrentLinkedDeque implementation had this.  Since this variable isn't public it isn't generated in the Javadoc, so I am not sure we need to fix this.